### DISCUSSION TITLE
Difficulty replicating Darcy Flow results (Fig 3) from FNO paper – Relative error stuck at 0.17

### DISCUSSION THREAD

**Discussion by lyyc199586** [neuraloperator/neuraloperator#628] [Q&A]
Hi all, 

I am trying to replicate the Darcy Flow results (Fig 3 Mid) from the original [FNO paper](http://arxiv.org/abs/2010.08895), where the reported relative error is around 10^-2:
<img width="1815" height="872" alt="image" src="https://github.com/user-attachments/assets/76349777-038a-4a21-b5cb-84fae5532c53" />

However, using the dataset and training script in the repo, my model only reaches about 0.17 (l2 loss):
<img width="755" height="376" alt="image" src="https://github.com/user-attachments/assets/aae4c37e-42ca-4de6-93dc-8f69d9cddb07" />

## My setup

FNO model
```python
# model setup
model = FNO(n_modes=(16, 16),
             in_channels=1,
             out_channels=1,
             hidden_channels=24,
             n_layers=4,
             fno_skip='linear',
             positional_embedding='grid',
             lifting_channel_ratio=2,
             projection_channel_ratio=2,
             factorization=None,
             )
model = model.to(device)

n_params = count_model_params(model)
print(f'\nOur model has {n_params} parameters.')
sys.stdout.flush()
```

Training:
```python
# model setup
n_epochs = 300
learning_rate = 5e-3
training_loss = "h1"
weight_decay = 1e-4
scheduler = "StepLR"
step_size = 60
gamma = 0.5
    
optimizer = AdamW(model.parameters(),
                                lr=learning_rate,
                                weight_decay=weight_decay)
# scheduler = torch.optim.lr_scheduler.CosineAnnealingLR(optimizer, T_max=50)
scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=step_size, gamma=gamma)

# loss functions
l2loss = LpLoss(d=2, p=2)
h1loss = H1Loss(d=2)

train_loss = h1loss
eval_losses={'h1': h1loss, 'l2': l2loss}


print('\n### MODEL ###\n', model)
print('\n### OPTIMIZER ###\n', optimizer)
print('\n### SCHEDULER ###\n', scheduler)
print('\n### LOSSES ###')
print(f'\n * Train: {train_loss}')
print(f'\n * Test: {eval_losses}')
sys.stdout.flush()
```

## Question:
Question:

- Could you kindly share the original dataset and exact config used in the FNO paper?

- Are there any key differences between the paper setup and the example in this repo that explain the gap?

Thanks in advance!
